### PR TITLE
[Workspace] Add functional tests for create/update workspace page

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

This PR adds functional tests for create/update workspace page.

### Issues Resolved

https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/1220

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/15811131/6aaa40df-3d9c-4815-b9a7-b677d512a0ee

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/15811131/1d85e0dc-e111-4e2b-9120-bdd154d99c5e

### CODE REVIEW

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      workspace.id = resp.body.result.id;
+    } else {
+      throw new Error(`Create workspace ${workspace} failed!`);
+    }
+  });
+});
+
+Cypress.Commands.add('checkWorkspace', (workspaceName, expected) => {
```
Comment (wanglam) [inline cypress/utils/dashboards/workspace-plugin/commands.js:56]: Shall we checkWorkspace based the workspace id? So we don't need to call list API and use workspace name to match worksapces.

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+              if (permissions && expected.permissions) {
+                const expectedPermissions = JSON.stringify(
+                  expected.permissions
+                );
+                const actualPermissions = JSON.stringify(permissions);
+                Object.entries(permissions).forEach(([key, value]) => {
+                  if (!expected.permissions[key]) {
+                    throw new Error(
+                      `permissions for workspace ${workspaceName} is not as expected, expected are:  ${expectedPermissions}, but are ${actualPermissions}`
```
Comment (wanglam) [inline cypress/utils/dashboards/workspace-plugin/commands.js:93]: Prefer using specific error message for permission key not exists.

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    );
+                  } else {
+                    if (expected.permissions[key].users) {
```
Comment (wanglam) [inline cypress/utils/dashboards/workspace-plugin/commands.js:96]: Shall we separate a string array compare function, then we can reused in users array and groups array. This function will check array length and all elements are equal.

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BASE_PATH } from '../../base_constants';
+import { WORKSPACE_API_PREFIX } from './constants';
+
+Cypress.Commands.add('deleteWorkspace', (workspaceName) => {
```
Comment (wanglam) [inline cypress/utils/dashboards/workspace-plugin/commands.js:9]: How about renaming to `deleteWorkspaceByName`? Maybe we will have `deleteWorkspaceById` in the future.

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    body: {
+      attributes: {
+        name: workspace.name,
+        description: 'test_description',
+      },
+    },
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      workspace.id = resp.body.result.id;
```
[the remark this answers is not in the document]
Comment (gaobinlong) [reply cypress/utils/dashboards/workspace-plugin/commands.js]: Sure, I've changed that, thanks!

File: cypress/utils/dashboards/workspace-plugin/commands.js (hunk 4 above)
Comment (gaobinlong) [reply cypress/utils/dashboards/workspace-plugin/commands.js]: I've changed that, and also added a new command `deleteWorkspaceById `.

File: cypress/utils/dashboards/workspace-plugin/commands.js (hunk 3 above)
Comment (gaobinlong) [reply cypress/utils/dashboards/workspace-plugin/commands.js]: I've added that functions, please help to take a look, thank you!

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/workspace_create.spec.js [hunk 6]
```diff
@@ -3,30 +3,186 @@
...
+    if (
+      Cypress.env('SAVED_OBJECTS_PERMISSION_ENABLED') &&
+      Cypress.env('SECURITY_ENABLED')
+    ) {
+      describe('Create a workspace with permissions successfully', () => {
+        before(() => {
+          cy.deleteWorkspaceByName(workspaceName);
+        });
+
+        it('should successfully create a worksapce with permissions', () => {
```
Comment (ruanyl) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/workspace_create.spec.js:125]: Typo `worksapce`

File: cypress/utils/dashboards/workspace-plugin/commands.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      return resp.body.result.id;
+    } else {
+      throw new Error(`Create workspace ${workspaceName} failed!`);
+    }
+  });
+});
+
+Cypress.Commands.add('checkWorkspace', (workspaceId, expected) => {
```
Comment (ruanyl) [inline cypress/utils/dashboards/workspace-plugin/commands.js:66]: Could you please leave comment to describe what does `checkWorkspace` do?

**Comment by wanglam** [comment]
Hi @gaobinlong , since the CI job failed. Could you upload the test successful videos?

File: cypress/utils/dashboards/workspace-plugin/commands.js (hunk 7 above)
Comment (gaobinlong) [reply cypress/utils/dashboards/workspace-plugin/commands.js:70]: Thanks, I've done that.

**Comment by gaobinlong** [comment]
> Hi @gaobinlong , since the CI job failed. Could you upload the test successful videos?

Thanks, I've added two videos, please help to take a look.

**Review by ruanyl — APPROVED** [review]

**Review by wanglam — APPROVED** [review]

**Comment by Hailong-am** [comment]
@gaobinlong would you mind fix this https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/8842528923/job/24281342558?pr=1221

``` Error: workspace yTd1vC is not as expected, expected features are: ["dashboards","visualize","discover","opensearchDashboardsOverview","workspace_update","workspace_overview"], but are: ["workspace_update","workspace_overview","dashboards","visualize","discover"]```

**Review by Hailong-am — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**